privacyscore/frontend/views.py

```diff
@@ -588,12 +589,30 @@ def scan_list_csv(request: HttpRequest, scan_list_id: int) -> HttpResponse:
 
 
 def site_result_json(request: HttpRequest, site_id: int) -> HttpResponse:
-    site = get_object_or_404(Site.objects.annotate_most_recent_scan_result(), pk=site_id)
-    scan_result = site.last_scan__result if site.last_scan__result else {}
+    if 'at' in request.GET:
+        # Check that the site even exists
+        site = get_object_or_404(Site, pk=site_id)
+
+        # TODO sanity check timestamp
+        try:
+            timestamp = datetime.strptime(request.GET['at'], "%Y-%m-%d")
+        except:
+            return render(request, 'frontend/site_result_json.html', {'site': site, 'highlighted_code': 'Incorrect timestamp format'})
+        try:
+            scan = Scan.objects.filter(site=site).filter(end__lte=timestamp).order_by('-id')[0]
```

Comment:
If we are ordering the results anyway, why not by the end attribute?
In addition, using .first() would probably be better than index-based access; you might get rid of the try-block that way.